Common/AR/source/LocationDataSource.cpp

```diff
@@ -0,0 +1,232 @@
+/*******************************************************************************
+ *  Copyright 2012-2019 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+
+#include "LocationDataSource.h"
+#include <QGeoPositionInfoSource>
+#include <QCompass>
+
+/*!
+  \class Esri::ArcGISRuntime::Toolkit::LocationDataSource
```

Comment:
The issue is the `LocationDataSource` class is not public in the Qt's Runtime API. And the AR's `LocationDataSource` don't need to be shared with the core runtime. For the AR point of view, it's just a wrapper to `QGeoPositionInfoSource` and `QCompass`, it's totally independent of the core runtime.

I think it's probably better to change the name of this class rather than using `QRTLocationDataSource` in this class.

Reply:
Are other Runtime APIs using the LocationDataSource from core or are they implementing a new toolkit version like what you did?

Reply:
They use the LocationDataSource from core but this class is public in other SDKs, not in Qt.